snuba/query/matchers.py

```diff
@@ -39,6 +39,12 @@ def expression(self, name: str) -> Expression:
         assert isinstance(ret, Expression)
         return ret
 
+    def optional_expression(self, name: str) -> Optional[Expression]:
```

Comment:
Since when you use these optional methods you are already checking whether the return type is None before using it, why not just having one `contains` method that returns True or False and then using the assert ones to get the value? 
Since you wrote the pattern, once you know the parameter was matched you can afford asserting that it was of the type you expected. So we would not have to duplicate all the accessor methods.